cassandra/src/main/scala/filodb.cassandra/columnstore/IngestionTimeIndexTable.scala

```diff
@@ -41,37 +36,40 @@ sealed class IngestionTimeIndexTable(val dataset: DatasetRef, val connector: Fil
                      |) WITH compression = {
                     'sstable_compression': '$sstableCompression'}""".stripMargin
 
-  def fromRow(row: Row): InfoRecord =
-    InfoRecord(row.getBytes("partition"), row.getBytes("info"))
-
-  val selectCql = s"SELECT partition, info FROM $tableString WHERE "
-  lazy val allPartReadCql = session.prepare(selectCql + "partition = ?")
-  lazy val inPartReadCql = session.prepare(selectCql + "partition IN ?")
+  lazy val allCql = session.prepare(
+      s"SELECT ingestion_time, start_time, info FROM $tableString " +
+      s" WHERE partition = ?")
+    .setConsistencyLevel(ConsistencyLevel.ONE)
 
   /**
-   * Retrieves all infos from a single partition.
-   */
-  def getInfos(binPartition: Array[Byte]): Observable[InfoRecord] = {
-    val it = session.execute(allPartReadCql.bind(toBuffer(binPartition)))
-                    .asScala.toIterator.map(fromRow)
-    Observable.fromIterator(it).handleObservableErrors
-  }
-
-  def getMultiInfos(partitions: Seq[Array[Byte]]): Observable[InfoRecord] = {
-    val query = inPartReadCql.bind().setList(0, partitions.map(toBuffer).asJava, classOf[ByteBuffer])
-    val it = session.execute(query).asScala.toIterator.map(fromRow)
-    Observable.fromIterator(it).handleObservableErrors
+    * Test method which returns all rows for a partition.
+    */
+  def readAllRows(partKeyBytes: ByteBuffer): ResultSet = {
+    session.execute(allCql.bind().setBytes(0, partKeyBytes))
   }
 
-  def scanInfos(tokens: Seq[(String, String)]): Observable[InfoRecord] = {
+  /**
+    * Returns Rows consisting of:
+    *
+    * partition:      ByteBuffer
+    * ingestion_time: long
+    * start_time:     long
+    * info:           ByteBuffer
+    *
+    * The ChunkSetInfo object contains methods for extracting fields from the info column.
+    */
+  def scanRowsByIngestionTime(tokens: Seq[(String, String)],
+                              ingestionTimeStart: Long,
+                              ingestionTimeEnd: Long): Iterator[Row] = {
```

Comment:
I dont see the benefit of using Iterator here. The memory utilization for the entire method is not async or lazy. 

Instead, it may be simpler to materialize to a `List` which will enable simpler `groupBy` semantics to the consumer of this API.

Reply:
Well, the number of rows received might be unbounded. An iterator makes more sense. Does C* driver read all rows from a query into memory up front, or does it do the right thing and stream as it goes?

Reply:
Agreed. ResultSet is indeed lazy